code/parse/parselo.cpp

```diff
@@ -1665,30 +1666,94 @@ int maybe_convert_foreign_character(int ch)
 }
 
 // Goober5000
-void maybe_convert_foreign_characters(char *line)
+// Yarn - The capacity of out must be at least the value returned by
+// get_converted_string_length(in) (plus one if add_null is true).
+// Returns the number of characters written to out.
+size_t maybe_convert_foreign_characters(char *in, char *out, bool add_null)
```

Comment:
Regarding 2: One of the two functions that calls this, process_raw_file_text in parselo.cpp, doesn't know the size of the output buffer, and there doesn't seem to be a way for it to get that size. What do you expect me to do (besides switch to SCP_string, which would probably entail numerous other changes that I'm hoping to avoid with this PR)?